### PR TITLE
Add null count test for apply_boolean_mask

### PR DESCRIPTION
**Pull request by harrism**
This PR adds a test to exercise the issue described in #6733. This issue was only reproduced on a laptop Pascal GPU, but I think it's a good test to have.  In summary, `copy_if`, used by `apply_boolean_mask` computes the output null count during as part of its custom scatter kernel, rather than using `cudf::count_unset_bits`.  #6733 describes an issue where the former is different from the latter. So it's good to have a test that verifies they get the same null count.

And since it's difficult to get a repro on a similar machine, this is a first step.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Review by karthikeyann — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,9 @@
 
 ## Bug Fixes
 
+ - PR #6903 Add null count test for apply_boolean_mask
+
+## Bug Fixes
```
Comment (kkraus14) [inline CHANGELOG.md:13]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 ## Bug Fixes
 
+ - PR #6903 Add null count test for apply_boolean_mask
```
Comment (kkraus14) [inline CHANGELOG.md:11]:
```suggestion
- PR #6903 Add null count test for apply_boolean_mask
```

**Review by karthikeyann — APPROVED** [review]
LGTM

**Comment by harrism** [comment]
rerun tests

**Review by devavret — APPROVED** [review]

**Pull request merged**